hypervisor/boot/sbl/abl_seed_parse.c

```diff
@@ -30,56 +30,49 @@ static void parse_seed_list_abl(void *param_addr)
 	uint32_t legacy_seed_index = 0U;
 	struct seed_info dseed_list[BOOTLOADER_SEED_MAX_ENTRIES];
 	struct dev_sec_info *sec_info = (struct dev_sec_info *)param_addr;
-
-	if (sec_info == NULL) {
-		goto fail;
-	}
-
-	if (sec_info->num_seeds < 2U ||
-			sec_info->num_seeds > ABL_SEED_LIST_MAX) {
-		goto fail;
-	}
-
-	/*
-	 * The seed_list from ABL contains several seeds which based on SVN
-	 * and one legacy seed which is not based on SVN. The legacy seed's
-	 * svn value is minimum in the seed list. And CSE ensures at least two
-	 * seeds will be generated which will contain the legacy seed.
-	 * Here find the legacy seed index first.
-	 */
-	for (i = 1U; i < sec_info->num_seeds; i++) {
-		if (sec_info->seed_list[i].svn <
-				sec_info->seed_list[legacy_seed_index].svn) {
-			legacy_seed_index = i;
+	bool parse_success = false;
+
+	if ((sec_info != NULL) && (sec_info->num_seeds >= 2U) && (sec_info->num_seeds <= ABL_SEED_LIST_MAX)) {
```

Comment:
is this ABL_SEED_LIST_MAX defined as 4?

I was told that this could be upgraded to 5 in future. 

what's the impact if you update this to 5? if no for now, we should change it to 5

Reply:
anyway, it should be fine if we still use 4 now

Reply:
I think we would do future things in future? since this commit is cherry-picked from master branch. we'd better to make an additional patch instead of modify this patch.

Reply:
ok, i would recommend that we should do this in future on "master" branch. (in case that if cse fw upgrades to max 5, then we would fail at that time)